google/cloud/bigquery/table.py

```diff
@@ -1075,6 +1086,41 @@ def external_catalog_table_options(
                 self._PROPERTY_TO_API_FIELD["external_catalog_table_options"]
             ] = value
 
+    @property
+    def foreign_type_info(self) -> Optional[_schema.ForeignTypeInfo]:
+        """Optional. Specifies metadata of the foreign data type definition in
+        field schema (TableFieldSchema.foreign_type_definition).
+
+        Returns:
+            Optional[schema.ForeignTypeInfo]:
+                Foreign type information, or :data:`None` if not set.
+
+        NOTE: foreign_type_info is only required if you are referencing an
+        external catalog such as a Hive table.
+        For details, see:
+        https://cloud.google.com/bigquery/docs/external-tables
+        https://cloud.google.com/bigquery/docs/datasets-intro#external_datasets
+        """
+
+        prop = self._properties.get(self._PROPERTY_TO_API_FIELD["foreign_type_info"])
```

Comment:
Even if we are exposing this at the table level, it needs to be fetched from schema, still, right?

This is exactly the sort of thing the `_get_sub_prop` (https://github.com/googleapis/python-bigquery/blob/54c8d07f06a8ae460c9e0fb1614e1fbc21efb5df/google/cloud/bigquery/_helpers.py#L781) and `_set_sub_prop` (https://github.com/googleapis/python-bigquery/blob/54c8d07f06a8ae460c9e0fb1614e1fbc21efb5df/google/cloud/bigquery/_helpers.py#L830) helpers are intended to be used for.

We even use it in other `Table` properties, such as `project`: https://github.com/googleapis/python-bigquery/blob/54c8d07f06a8ae460c9e0fb1614e1fbc21efb5df/google/cloud/bigquery/table.py#L175-L177


```suggestion
        prop = _helpers._get_sub_prop(self._properties, self._PROPERTY_TO_API_FIELD["foreign_type_info"])
```

Reply:
Done.

Reply:
Complete.